### PR TITLE
Update documentation

### LINKED ISSUE #840: Improve Job worker section
https://stage.docs.zeebe.io/bpmn-workflows/data-flow.html

"Tight coupling" often has negative connotations, so users may want to avoid it instinctively. It is valid and sometimes better usage, though. Let's try to find better terms to distinguish the use cases.

### PR DESCRIPTION
**Pull request by saig0**
closes #840

### CODE REVIEW

File: docs/src/basics/job-workers.md [hunk 1]
```diff
@@ -4,33 +4,40 @@ A job worker is a component capable of performing a particular step in a workflo
...
 
-* **Type**: Describes the work item and is defined in each task in the workflow model. The type is referenced by workers to request the jobs they are able to perform.
-* **Payload**: The business data the worker should work on in the form of a JSON document/object.
-* **Headers**: Additional metadata defined in the workflow, e.g. describing the structure of the payload.
+* **Type**: Describes the work item and is defined in each task in the workflow. The type is referenced by workers to request the jobs they are able to perform.
+* **Variables**: The contextual/business data of the workflow instance that is required by the worker to do its work.
+* **Custom Headers**: Additional static metadata defined in the workflow. Mostly used to configure a worker which is used for more than one workflow step.
 
-## Requesting Jobs From The Broker
+## Requesting Jobs from the Broker
```
Comment (Zelldon) [inline docs/src/basics/job-workers.md]: I think it was intended that all words begin with an upper case letter :sweat_smile:

File: docs/src/basics/job-workers.md [hunk 2]
```diff
@@ -4,33 +4,40 @@ A job worker is a component capable of performing a particular step in a workflo
...
 ![zeebe-job-workers-requesting-jobs](/basics/zeebe-job-workers-graphic.png)
 
-Many workers can request the same job type in order to scale up processing. In this scenario, the Zeebe broker ensures that each job is sent to only one of the workers. It can also reassign a job in case a worker is unable to complete it.
+Many workers can request the same job type in order to scale up processing. In this scenario, the broker ensures that each job is sent to only one of the workers.
+
+On requesting jobs, the following properties can be set:
+
+* **Worker**: The identifier of the worker. Used for auditing purposes.
+* **Timeout**: The time a job is assigned to the worker. If a job is not completed within this time then it can be requested again from a worker.
+* **Amount**: The maximum number of jobs which are requested.
```
Comment (Zelldon) [inline docs/src/basics/job-workers.md]: https://github.com/zeebe-io/zeebe/issues/1894 -> maxJobsActive

File: docs/src/java-client-examples/README.md [hunk 3]
```diff
@@ -25,7 +25,7 @@ Import the Maven project in the `samples` directory into your IDE to start hacki
 
 ## Data
 
-* [Handle the payload as POJO](java-client-examples/data-pojo.html)
+* [Handle variables as POJO](java-client-examples/data-pojo.html)
```
Comment (Zelldon) [inline docs/src/java-client-examples/README.md]: Will you create an issue for the variable pojo handling?

File: docs/src/basics/partitions.md [hunk 4]
```diff
@@ -23,3 +23,11 @@ A partition is a persistent append-only event stream. Initially, a partition is
 ## Replication
 
 For fault tolerance, data in a partition is replicated from the leader of the partition to its *followers*. Followers are other Zeebe Broker nodes that maintain a copy of the partition without performing event processing.
+
+## Recommendations
+
+Choosing the number of partitions depends on the use case, workload and cluster setup. Here are some rules of thumb:
+
+* For testing and early development, start with a single partition. Note that Zeebe's workflow processing is highly optimized for efficiency, so a single partition can already handle high event loads.
+* With a single Zeebe Broker, a single partition is mostly enough. However, if the node has many cores (> 4) then more partitions can increase the total throughput.
```
Comment (Zelldon) [inline docs/src/basics/partitions.md]: And thread count

**Review by ChrisKujawa — CHANGES_REQUESTED** [review]

File: docs/src/basics/job-workers.md (hunk 1 above)
Comment (saig0) [reply docs/src/basics/job-workers.md]: On other places, we don't start "filler" words with an upper case. I think it's ok ;)

Comment (Zelldon) [reply docs/src/basics/job-workers.md]: ok

**Comment by saig0** [comment]
@Zelldon , please have a look at the latest commit.

**Review by ChrisKujawa — APPROVED** [review]

**Comment by saig0** [comment]
bors retry

**Pull request merged**